flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/com/ververica/cdc/connectors/mysql/utils/MySqlTypeUtils.java

```diff
@@ -0,0 +1,236 @@
+                return DataTypes.BOOLEAN();
+            case TINYINT:
+                // MySQL haven't boolean type, it uses tinyint(1) to represents boolean type
+                // user should not use tinyint(1) to store number although jdbc url parameter
+                // tinyInt1isBit=false can help change the return value, it's not a general way
+                // btw: mybatis and mysql-connector-java map tinyint(1) to boolean by default
+                return column.length() == 1 ? DataTypes.BOOLEAN() : DataTypes.TINYINT();
+            case TINYINT_UNSIGNED:
+            case TINYINT_UNSIGNED_ZEROFILL:
+            case SMALLINT:
+                return DataTypes.SMALLINT();
+            case SMALLINT_UNSIGNED:
+            case SMALLINT_UNSIGNED_ZEROFILL:
+            case INT:
+            case INTEGER:
+            case MEDIUMINT:
+            case YEAR:
+                return DataTypes.INT();
+            case INT_UNSIGNED:
+            case INT_UNSIGNED_ZEROFILL:
+            case INTEGER_UNSIGNED:
+            case INTEGER_UNSIGNED_ZEROFILL:
+            case MEDIUMINT_UNSIGNED:
+            case MEDIUMINT_UNSIGNED_ZEROFILL:
+            case BIGINT:
+                return DataTypes.BIGINT();
+            case BIGINT_UNSIGNED:
+            case BIGINT_UNSIGNED_ZEROFILL:
+            case SERIAL:
+                return DataTypes.DECIMAL(20, 0);
+            case FLOAT:
+            case FLOAT_UNSIGNED:
+            case FLOAT_UNSIGNED_ZEROFILL:
+                return DataTypes.FLOAT();
+            case REAL:
+            case REAL_UNSIGNED:
+            case REAL_UNSIGNED_ZEROFILL:
+            case DOUBLE:
+            case DOUBLE_UNSIGNED:
+            case DOUBLE_UNSIGNED_ZEROFILL:
+            case DOUBLE_PRECISION:
+            case DOUBLE_PRECISION_UNSIGNED:
+            case DOUBLE_PRECISION_UNSIGNED_ZEROFILL:
+                return DataTypes.DOUBLE();
+            case NUMERIC:
+            case NUMERIC_UNSIGNED:
+            case NUMERIC_UNSIGNED_ZEROFILL:
+            case FIXED:
+            case FIXED_UNSIGNED:
+            case FIXED_UNSIGNED_ZEROFILL:
+            case DECIMAL:
+            case DECIMAL_UNSIGNED:
+            case DECIMAL_UNSIGNED_ZEROFILL:
+                return column.length() <= 38
+                        ? DataTypes.DECIMAL(column.length(), column.scale().orElse(0))
+                        : DataTypes.STRING();
+            case TIME:
+                return column.length() >= 0 ? DataTypes.TIME(column.length()) : DataTypes.TIME();
+            case DATE:
+                return DataTypes.DATE();
+            case DATETIME:
+                return column.length() >= 0
+                        ? DataTypes.TIMESTAMP(column.length())
+                        : DataTypes.TIMESTAMP();
+            case TIMESTAMP:
+                return column.length() >= 0
+                        ? DataTypes.TIMESTAMP_LTZ(column.length())
+                        : DataTypes.TIMESTAMP_LTZ();
```

Comment:
DataTypes.TIMESTAMP() will be of TimestampType.DEFAULT_PRECISION, which is 6, we should set to 0 to be consistent with DebeziumSchemaDataTypeInference#infer method.